llvm-external-projects/iree-dialects/include/iree-dialects/Transforms/Listener.h

```diff
@@ -42,7 +42,7 @@ struct RewriteListener {
   /// Notification handler for when the specified operation is about to be
   /// replaced with another set of operations. This is called before the uses of
   /// the operation have been replaced with the specific values.
-  virtual void notifyOperationReplaced(Operation *op, ValueRange newValues) {}
+  virtual void notifyRootReplaced(Operation *op, ValueRange newValues) {}
```

Comment:
Why rename this to `root`?

Reply:
Because the upstream name changed and I am adapting everything accordingly (since CSE and canonicalize are copies with listeners).